packages/nx/src/native/walker.rs

```diff
@@ -176,6 +176,7 @@ where
     let mut walker = WalkBuilder::new(&directory);
     walker.require_git(false);
     walker.hidden(false);
+    walker.parents(false);
     walker.git_ignore(use_ignores);
     if use_ignores {
         walker.add_custom_ignore_filename(".nxignore");
```

Comment:
```suggestion
    walker.git_ignore(use_ignores);
    if use_ignores {
        // disable the parent gitignore if the directory is a git repo, otherwise use the parent gitignore
        walker.parents(!directory.join(".git").exists());
        walker.add_custom_ignore_filename(".nxignore");
```